### PR TITLE
added ip selector code

### PR DESCRIPTION
**Pull request by rm-Umar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +150,10 @@ other technologies change the origin of a request that a web server can see.
   This strategy simply looks at the IP of a request and uses that to extract
   more data using IPinfo.
 
+- `XForwardedFor`
```
Comment (UmanShahzad) [inline README.md]: We should have a new section for this the way we have for other strategies, and it should be named accordingly, like `XForwardedForIPStrategy`.

File: README.md [hunk 2]
```diff
@@ -150,6 +150,10 @@ other technologies change the origin of a request that a web server can see.
...
+  This strategy will extract IP from `X-Forwarded-For` header and if it's null
+  it'll extract IP using `REMOTE_ADDR` of client.
```
Comment (UmanShahzad) [inline README.md]:
```suggestion
  This strategy will extract the IP from the `X-Forwarded-For` header and if it's null
  it'll extract IP using `REMOTE_ADDR` of client.
```

File: src/main/java/io/ipinfo/spring/strategies/ip/XForwardedFor.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class XForwardedFor implements IPStrategy {
+    /**
+     * Gets the ip address using the X-Forwarded-For header.
+     *
+     * @param request originated from a client to this server.
+     * @return
+     */
+    @Override
+    public String getIPAddress(HttpServletRequest request) {
+        String XforwardedFor = request.getHeader("X-Forwarded-For");
```
Comment (UmanShahzad) [inline src/main/java/io/ipinfo/spring/strategies/ip/XForwardedFor.java]: I'm not sure myself, but does this lowercase the headers properly? Like does it handle mixed case on its own?

Comment (UmanShahzad) [reply src/main/java/io/ipinfo/spring/strategies/ip/XForwardedFor.java]: Also the variable should be lowercase, `xForwardedFor`, I think.

Comment (rm-Umar) [reply src/main/java/io/ipinfo/spring/strategies/ip/XForwardedFor.java]: Yes the the function is case insensitive so it'll handle all it properly

File: README.md (hunk 1 above)
Comment (rm-Umar) [reply README.md]: The main section is `IPStrategy` and `SimpleIPStrategy` and `XForwardedForIPStrategy` will go under that section as bullet points to match the rest of the document format

Comment (UmanShahzad) [reply README.md]: Oh right.

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**